File: apps/android/feature/ai/domain/src/main/java/com/scribblefit/feature/ai/domain/engine/SyncRepository.kt
package com.scribblefit.feature.ai.domain.engine

import com.scribblefit.feature.ai.domain.model.ParsedWorkout
import com.scribblefit.feature.ai.domain.model.SyncItem
import com.scribblefit.feature.ai.domain.model.SyncStatus
import kotlinx.coroutines.flow.Flow

interface SyncRepository {
    fun getPendingSyncItems(): Flow<List<SyncItem>>
    fun getAllSyncItems(): Flow<List<SyncItem>>
    suspend fun updateSyncStatus(id: String, status: SyncStatus)
    suspend fun saveParsedWorkout(syncItemId: String, workout: ParsedWorkout)
    suspend fun enqueueScribble(id: String, rawText: String)
    suspend fun deleteSyncItem(id: String)
    suspend fun saveFeedItem(
        id: String,
        type: String,
        jsonData: String,
        status: SyncStatus = SyncStatus.COMPLETED
    )
    suspend fun syncWorkouts()
}

File: apps/ios/ScribbleFit/Sources/Core/AI/SyncRepository.swift
import Combine
import Foundation

public protocol SyncRepository: Sendable {
    func getPendingSyncItems() async throws -> [AISyncItem]
    func getAllSyncItems() async throws -> [AISyncItem]
    func observeAllSyncItems() -> AnyPublisher<[AISyncItem], Never>
    func updateSyncStatus(id: String, status: SyncStatus) async throws
    func saveParsedWorkout(syncItemId: String, workout: ParsedWorkout) async throws
    func enqueueScribble(id: String, rawText: String) async throws
    func saveFeedItem(id: String, type: String, jsonData: String, status: SyncStatus) async throws
    func deleteSyncItem(id: String) async throws
    func syncWorkouts() async throws
}
